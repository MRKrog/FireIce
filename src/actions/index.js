export const setHouses = (houses) => ({
  type: "SET_HOUSES",
  houses
})

export const setLoading = (loading) => ({
  type: "SET_LOADING",
  loading
})
