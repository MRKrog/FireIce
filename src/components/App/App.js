import React, { Component } from 'react';
import logo from './logo.svg';
import { connect } from 'react-redux';
import * as actions from '../../actions/index';
import CardContainer from '../CardContainer/CardContainer';
import Loading from '../Loading/Loading';

export class App extends Component {

  componentDidMount(){
    this.props.setLoading(true)
    this.getPeople()
  }

  getPeople = async () => {
    try {
      const url = "http://localhost:3001/api/v1/houses";
      const response = await fetch(url)
      if(!response.ok) { throw new Error('Bad Fetch')}
      const data = await response.json()
      this.props.setHouses(data)
      this.props.setLoading(false)
    } catch (error) {
        console.log(error.message);
    }
  }


  render() {
    const { houses, loading } = this.props
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        <div className='Display-info'>
        {
          !loading ?
          <CardContainer houses={houses}/> :
          <Loading />
        }
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  houses: state.houses,
  loading: state.loading
})

export const mapDispatchToProps=(dispatch) => ({
  setHouses: (data) => dispatch(actions.setHouses(data)),
  setLoading: (data) => dispatch(actions.setLoading(data)),
})


export default connect(mapStateToProps, mapDispatchToProps)(App)
