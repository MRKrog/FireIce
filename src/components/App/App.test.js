import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App'
import { shallow } from 'enzyme';

import * as actions from '../../actions/index'

const mockHouses = [
  { name: "House Corbray of Heart's Home" },
  { name: "House Dayne of Starfall" },
  { name: "House Hightower of the Hightower" }
]


describe('App', () => {

  describe('App Component', () => {
    let wrapper;
    let mockSetLoading = jest.fn()
    let mockSetHouses = jest.fn()
    let mockLoading = false;

    beforeEach(() => {
      wrapper = shallow(<App loading={mockLoading}
                             houses={mockHouses}
                             setLoading={mockSetLoading}
                             setHouses={mockSetHouses}
                             />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {

    it('should return an object with the houses array', () => {
      const mockState = {
        houses: mockHouses,
        filter: 'SET_HOUSES'
      }
      const expected = {
        houses: mockHouses
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('should return an object with the loading status', () => {
      const mockState = {
        loading: false,
        filter: 'SET_LOADING'
      }
      const expected = {
        loading: false
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

  })

  describe('mapDispatchToProps', () => {

    it('calls dispatch with an setHouses action when action.setHouses is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = actions.setHouses(mockHouses)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setHouses(mockHouses)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with an setLoading action when action.setLoading is called', () => {
      let mockLoading = false;
      const mockDispatch = jest.fn()
      const actionToDispatch = actions.setLoading(mockLoading)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setLoading(mockLoading)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })

})
