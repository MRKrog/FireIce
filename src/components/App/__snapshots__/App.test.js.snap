// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to Westeros
    </h2>
  </div>
  <div
    className="Display-info"
  >
    <CardContainer
      houses={
        Array [
          Object {
            "name": "House Corbray of Heart's Home",
          },
          Object {
            "name": "House Dayne of Starfall",
          },
          Object {
            "name": "House Hightower of the Hightower",
          },
        ]
      }
    />
  </div>
</div>
`;
